### ISSUE TITLE
Local Mining Pool Not Distributing New Work (Same Hash)

### ISSUE THREAD

**Issue by Thomas-Veil** [iron-fish/ironfish#3839]
### What happened?

Linux fresh install from 1.0.0 updated to 1.0.1

Local miner running on the node receives the new work (new hash) the pool running on the node is not distributing the new work correctly to the miners or the remote miners are not receiving it correctly from the pool. All the remote miners are linux all fresh installs 1.0.0 upgraded to 1.0.1.

New Work: 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64

It's always the same hash.

This problem also occurs on the official pool (pool.ironfish.network).

### Version

1.0.1

### Debug output

```shell
Pool (pool.ironfish.network) / Exactly the same on my local pool:

ironfish miners:start -v -t 1 -p pool.ironfish.network -a X

new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25657 4.98 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25658 4.96 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25659 4.91 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25660 4.98 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25661 4.97 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25662 4.96 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25663 4.87 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25664 4.94 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25665 4.96 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25666 4.96 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25667 4.96 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25668 4.95 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25669 4.94 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25670 4.94 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25671 4.95 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25672 4.92 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25673 4.95 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25674 4.96 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25675 4.95 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25676 4.95 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25677 4.97 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25678 4.93 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25679 4.92 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25680 4.98 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25681 4.96 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25682 4.93 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25683 4.96 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25684 4.98 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25685 4.96 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25686 4.96 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25687 4.98 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25688 4.99 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25689 4.93 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25690 4.89 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25691 4.97 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25692 4.98 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25693 4.86 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25694 4.95 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25695 4.97 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25696 4.97 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25697 4.97 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25698 5.01 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25699 4.98 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25700 4.89 MH/s
new work 00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64 25701 4.94 MH/s

Local:

ironfish miners:start -v -t 1

new work 000000000000003aafc76a8c268fa79dd83aeec9a15910ae8d52fa8d9cdf0ae5, 1 1.23 MH/s
new work 000000000000003ab71b8e707efe5d422e88eb313376a60562bb396292d7f620, 2 1.22 MH/s
new work 000000000000003a8b306c4879440598439196564e1bc82ff12827f372a24a5f, 3 1.24 MH/s
new work 000000000000003a927ffe7178136d32c3ba5b09f87030d9318b950e13c7e8c3, 4 1.24 MH/s
new work 000000000000003a99d164448bdfe9016154c12920f87ff8b14a33f1cab6ed2d, 5 1.23 MH/s
new work 000000000000003aa1249e71147157ebaeb2d34a5973909c6f0dd65161160c57, 6 1.23 MH/s
new work 000000000000003aa879ada6c94a74241b4accfc0875c1f69906a97c3190d9b7, 7 1.24 MH/s
new work 000000000000003aafd09295b9dfb5a9a2ed1e81fef8cd6712caeeef5672457e, 8 1.22 MH/s
new work 000000000000003a8b34139cfacd2eaa9fe45a75b0dc053b0f6ee4600a023a38, 9 1.23 MH/s
new work 000000000000003a9282bca00d4592e681ec050cf58bad6d0efa412bd5ed8bfe, 10 1.23 MH/s
new work 000000000000003a66b3e32efd6ed60b92390a4c57639e673e5bf1b0c52d72fe, 11 1.23 MH/s
new work 000000000000003a6dfd151271946e939431a729f517b9a96f08f78d0497e232, 12 1.24 MH/s
new work 000000000000003a75481893c363f0c97594a101eac3d051cc2c168dbdb50491, 13 1.22 MH/s
new work 000000000000003a7c94ee61627496965d0f9e895e4183191cf975e1de2c8fcd, 14 1.23 MH/s
new work 000000000000003a83e3972a158a7fe3a5dced7cd3ef9430c83729e4d6b342e5, 15 1.23 MH/s
new work 000000000000003a5f67f6303768114e128c6fc84bd231b68d65ec7536944326, 16 1.23 MH/s
new work 000000000000003a3b0314438d2fdfdd15e4ba01299253197cbb0ffe44118e1e, 17 1.23 MH/s
new work 000000000000003a4246d2c6d3fe1c3d24111969955f3d02a77eda269bbd6434, 18 1.24 MH/s
```

**Comment by hughy** [comment]
It looks like the remote pool is not sending new work, but your local pool _is_ sending new work

I'm looking into the pool at pool.ironfish.network

**Comment by Thomas-Veil** [comment]
Problem still exists with latest version (1.2.0).

**Comment by Thomas-Veil** [comment]
Same (1.3.0).

Running a local pool. Remote miners are not receiving new work correctly or the local pool is not sending the new work correctly. it's always the same hash:
00000000494cff9a3f4f473f91d116af7382c45e653facfeef85b8f43d9d6b64

**Comment by Thomas-Veil** [comment]
Same (1.4.0).

**Comment by hughy** [comment]
This is a logging issue that should be resolved by #3974 in the next release.
